[clang][ptrauth] Add support for querying the ptrauth schema of a type (#138482)
llvm/llvm-project: clang/docs/PointerAuthentication.rst

significant; in particular, some implementations are known to not leave
meaningful data in the low bits.

``__ptrauth type queries``
^^^^^^^^^^^^^^^^^^^^^^^^^^

There are a number of builtins that can be used to query the ptrauth qualifier

> parameters of a type, including those configured implicitly. These are:

Reviewer: ```suggestion
arguments for a type, including those configured implicitly. These are:
```

> .. code-block:: c
> __builtin_ptrauth_has_authentication(type)
> __builtin_ptrauth_schema_key(type)
> __builtin_ptrauth_schema_is_address_discriminated(type)
> __builtin_ptrauth_schema_extra_discriminator(type)
> __builtin_ptrauth_schema_options(type)

Reviewer: ```suggestion
.. code-block:: c

  __builtin_ptrauth_has_authentication(type)
  __builtin_ptrauth_schema_key(type)
  __builtin_ptrauth_schema_is_address_discriminated(type)
  __builtin_ptrauth_schema_extra_discriminator(type)
  __builtin_ptrauth_schema_options(type)
```
This should unbreak the documentation build, I expect.

> All these builtins are compile time constants. The schema queries are only valid

Reviewer: ```suggestion
All these builtins are usable within a constant expression. The schema queries are only valid
```
